### ISSUE TITLE
org-modern-mode breaks org-indent-mode

### ISSUE THREAD

**Issue by zealotrush** [minad/org-modern#7]
Hi, first of all, thanks for the great package.

I found that the `org-modern-mode` breaks the `org-indent-mode`.  Here is a screenshot of an org-mode buffer with `org-indent-mode` being `t` and `org-modern-mode` being `nil`.

<img width="559" alt="image" src="https://user-images.githubusercontent.com/2275580/155916791-e9792c3d-45a6-4232-ac17-8471146dfb84.png">

Then if I set `org-modern-mode` to `t`, it becomes this.  Note that the indentation disappeared (`org-indent-mode` is still `t`).

<img width="559" alt="image" src="https://user-images.githubusercontent.com/2275580/155916872-4b8e181b-07f8-4bcf-b7d5-c694b729cb78.png">

More strangely, when I edit the buffer, the indentation got on and off unpredictably.

<img width="559" alt="image" src="https://user-images.githubusercontent.com/2275580/155917134-fce39708-3081-492a-8445-25e3e9484a77.png">

Is this a bug?

**Comment by Jackson-soft** [comment]
same question

**Comment by minad** [comment]
Yes, this is a known issue. org-indent-mode is currently incompatible with org-modern-mode. This is wontfix for now since org-indent-mode adjusts the line-prefix text properties which are also used by org-modern-mode. Not sure if it is possible to make this work. To be honest, I consider org-indent-mode a waste of space and not modern at all. ;)

**Comment by 404cn** [comment]
block's indent guide line also didn't show in org-indent-mode, I choose disable org-indent-mode.

**Comment by minad** [comment]
@404cn Yes, org-indent-mode is quite a problematic. It is an intrusive mode which breaks all kinds of other modes. I never used org-indent-mode since it always felt a bit unpredictable. Therefore it is okay if org-modern cannot be made to work with it. I will still keep this issue open in case someone finds a solution. But for example the nice org-modern source blocks use the fringe. If you indent the block, it won't look good.

**Comment by minad** [comment]
For reference, here is a heated discussion on how horrible org-indent-mode is and how org lives in the dark ages without org-indent-mode: https://github.com/syl20bnr/spacemacs/issues/2732 😆

**Comment by minad** [comment]
I fiddled a bit around and made `org-modern-mode` slightly less intrusive. But it seems impossible to make these modes cooperate nicely in all scenarios, since `org-indent-mode` is quite a bulldozer. I recommend to turn of `org-indent-mode` if you want to use `org-modern-mode`.

**Comment by stefan2904** [comment]
appreciate the effort.

do you think it is possible to build something like a modern org-indent-mode?

**Comment by minad** [comment]
I think it should be possible to make org-modern-mode more compatible with org-indent-mode by not using the fringe for source blocks. It needs a bit of work. Currently org-modern-mode doesn't aim for full compatibility. You can use some of the org-modern style elements with org-indent-mode without issues, but not all.

**Comment by trev-dev** [comment]
Perhaps if org-modern-mode implemented its own visual solution to level indentation? The reason I'd ask for this is because the indentation level is how I easily appreciate how many levels deep I actually am.

I don't have to think about what symbol precedes the text, or what color that headline is and try to remember 6 variations thereof and eventually surrender to un-hiding the leading asterisks.

**Comment by jdtsmith** [comment]
I quite like org-modern, and I quite like org-indent-mode.  So I co-opted the latter's use of line-prefix and wrap-prefix to approximate the block-styling of the former:

  https://github.com/jdtsmith/org-modern-indent/

Works well for me; happy for any feedback.  I'm not a deep org user, so I'm not sure what else org-indent-mode currently breaks with the workarounds @minad implemented.

**Comment by minad** [comment]
@jdtsmith The kind of cooperation you implemented between indent mode and modern mode is nice. There are two downsides of your approach:

1. It does not handle line-spacing>0 well. I usually increase line-spacing since I like if Emacs doesn't look so dense. It should be possible to fix this with some display/image trickery.
2. Using prefix characters indents the source blocks. For non-indent users this is not so nice since everything appears shifted. By using the fringe, this problem doesn't occur. For indent users this is of course not an issue.

For these two reasons I went with the current fringe solution.

**Comment by jdtsmith** [comment]
Thanks @minad.  

1. Good point, I hadn't thought of line spacing. Some fonts have more vertical breathing room, so that could be a workaround.
1. This is definitely _only_ for users of org-indent.  Without indent enabled, the fringe solution already in modern is clearly superior (and saves space).

**Comment by minad** [comment]
> 1. Good point, I hadn't thought of line spacing. Some fonts have more vertical breathing room, so that could be a workaround.

That's not a good workaround. We shouldn't restrict ourselves to only such fonts. As I wrote, the only proper solution is to use display trickery.

> 2.This is definitely _only_ for users of org-indent.  Without indent enabled, the fringe solution already in modern is clearly superior (and saves space).

Sure. My second point only explains why I use the fringe. I had tried inserting characters as alternative and I didn't like it. However I think the fringe solution only looks nice if you use large frame borders and window dividers as in Nicolas Rougier's Nano themes. 

Anyway, if we figure out a proper solution for 1. we can add this to org-modern itself, such that org-modern is more compatible with org-indent. It would be nice if we can achieve consistent looks for both cases (fringe vs non-fringe).

**Comment by ParetoOptimalDev** [comment]
> how org lives in the dark ages without org-indent-mode: [syl20bnr/spacemacs#2732](https://github.com/syl20bnr/spacemacs/issues/2732)

I actually had to begrudingly choose org-indent-mode over org-modern-mode just now for this incompatibility. I'll have to try @jdtsmith's package later and evaluate if I'm okay with no line-spacing everywhere.

Dark ages might be a little much, but the hierarchy that org-indent-mode adds is probably one of the things that makes org-mode most understandable to me.

**Comment by Andre0991** [comment]
Just FYI, strangely, I cannot reproduce this issue. The indentation is kept when using `org-modern`.

My (non minimal) org config:

```emacs-lisp
(use-package org
  :defer t
  :custom
  (org-startup-indented t)
  (org-confirm-babel-evaluate nil)
  (org-special-ctrl-a/e t)
  (org-hide-emphasis-markers t)
  (org-pretty-entities t)
  (org-ellipsis "…"))

(use-package org-modern
  :init
  (add-hook 'org-mode-hook #'org-modern-mode))
```

Emacs and org versions:
```
GNU Emacs 29.0.50 (build 1, x86_64-apple-darwin19.6.0, NS appkit-1894.60 Version 10.15.7 (Build 19H2)) of 2022-07-24

Org mode version 9.5.4 (release_9.5.4-3-g6dc785 @ /usr/local/Cellar/emacs-plus@29/29.0.50/share/emacs/29.0.50/lisp/org/)
```

**Comment by tshu-w** [comment]
The indentation does exist, but there is still a problem with the indentation at the body (compare org-super-star).
See level 1, level 2, level 3 in the screenshot, they should be aligned with the heading.
<img width="212" alt="Screen Shot 2022-08-04 at 15 19 25" src="https://user-images.githubusercontent.com/13161779/182787953-00422e57-0237-41c7-960c-225dd411f586.png">

**Comment by Andre0991** [comment]
Got it. Indeed, it's misaligned here too. I hadn't noticed because it was less apparent on the other file I was working on.
